app/lib/three_scale/analytics/user_tracking.rb

```diff
@@ -30,12 +30,20 @@ def method_missing(method_sym, *args, &block)
           end
         end
 
-        def segment_configured?
-          defined?(::Segment)
+        def self.segment_configured?
+          Features::SegmentConfig.enabled? && defined?(::Segment)
         end
 
         def initialize(config = {})
-          @adapter = segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
+          @config = config
+        end
+
+        private
+
+        attr_reader :config
+
+        def adapter
+          @adapter ||= TrackingAdapter.segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
```

Comment:
@Martouta @hallelujah Since the only place that instantiate the is the constant assignment (https://github.com/3scale/porta/pull/903/files#diff-d32b184052c26bb111d811f12f0a2e9aR51), I think we do not need the adapter to be a class method. Making the @adapter an instance variable we can memoize it, since `Segment` will only have one instance of `TrackingAdapter`.

What do you folks think? 🤔

Reply:
@duduribeiro I agree that `adapter` needs to be an instance method because it relies on `config` 😄